src/main/java/com/getindata/connectors/http/sink/HttpSinkWriter.java

```diff
@@ -0,0 +1,83 @@
+package com.getindata.connectors.http.sink;
+
+import com.getindata.connectors.http.SinkHttpClient;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.api.connector.sink2.Sink;
+import org.apache.flink.connector.base.sink.writer.AsyncSinkWriter;
+import org.apache.flink.connector.base.sink.writer.BufferedRequestState;
+import org.apache.flink.connector.base.sink.writer.ElementConverter;
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.groups.SinkWriterMetricGroup;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Sink writer created by {@link HttpSink} to write to an HTTP endpoint.
+ *
+ * <p>More details on the internals of this sink writer may be found in {@link AsyncSinkWriter}
+ * documentation.
+ *
+ * @param <InputT> type of the elements that should be sent through HTTP request.
+ */
+@Slf4j
+public class HttpSinkWriter<InputT> extends AsyncSinkWriter<InputT, HttpSinkRequestEntry> {
+  private final String endpointUrl;
+  private final SinkHttpClient sinkHttpClient;
+  private final SinkWriterMetricGroup metrics;
```

Comment:
It seems that this does not have to be a field.
In constructor you do 
```
 this.metrics = context.metricGroup();
    this.numRecordsSendErrorsCounter = metrics.getNumRecordsSendErrorsCounter();
```
and "metrics" field is not used anywhere else in this class.